peer/pendingheap/list.go

```diff
@@ -69,17 +69,10 @@ func New(transport peer.Transport, opts ...ListOption) *List {
 			&pendingHeap{},
 			plOpts...,
 		),
-		capacity: cfg.capacity,
 	}
 }
 
 // List is a PeerList which rotates which peers are to be selected in a circle
 type List struct {
 	*peerlist.List
-	capacity int
-}
-
-// Capacity is the maximum number of peers the peer list will hold.
```

Comment:
Clever, though I'm unable to get this working. I believe you can only add methods within the same package. 

(I'm kind of sad though, this would have been really cool)